### ISSUE TITLE
Trouble Opening on M1 Mac

### ISSUE THREAD

**Issue by thribbyribby** [gamma-02/NBT_to_CSV#2]
Hi, I've got a problem that I suspect is much more on my end than it is on your end, but I can't seem to open the .jar file on my M1 macbook air. I tried opening it through terminal commands (java -jar [directory to the jar file]), but I got some error message that I really just don't understand.

I've attached the output from my terminal shell, and here's an error from the Console app that kept popping up as well: "Sandbox: com.apple.quicklook.ThumbnailsAg(58772) deny(1) file-read-xattr /Users/COMPUTER_NAME/Downloads/nbttocsv-1.1-SNAPSHOT.jar"
[terminal.shell.output.txt](https://github.com/user-attachments/files/17219248/terminal.shell.output.txt)

I'm totally out of my depth when it comes to programming and all this, and was wondering if you could help point me in the right direction? 
[terminal.shell.output.txt](https://github.com/user-attachments/files/17219248/terminal.shell.output.txt)

**Comment by gamma-02** [comment]
Hello! I compiled the release versions of NBT to CSV on an m2 mac, which seemed to fix errors relating to M1 mac
However.
they obviously did not.
I suggest you compile from source yourself, from your laptop. That should bypass any weird platform issues.

I can't do anything to fix this at the moment, unfortunately. The issue stems from JavaFX, the UI library I use for the application, and if you were to compile from source on your machine, JFX will magically become platform agnostic like _Java_ is supposed to be. Sorry!

Gamma

**Comment by thribbyribby** [comment]
Sorry to come back with another problem, especially since this is definitely on my end since I have no programming experience, but, how do I compile it from source? I've downloaded the source code zip, and opened that, but from there I don't know where to go.

I've followed tutorial videos telling me to use the terminal commands "javac [file name].java" and "cat [file name].java", but I don't know which .java file to use the commands with from your source files. Each one I've tried has given me several errors. 

Again, sorry to keep coming to you with my tech-illiteracy problems, but I figured that you would know how to compile your files a lot better than years-old forum posts and youtube videos.
